src/pages/organize/[orgId]/projects/[campId]/emails/[emailId]/insights.tsx

```diff
@@ -0,0 +1,153 @@
+import Head from 'next/head';
+import { Paper, Typography } from '@mui/material';
+import { Box, useTheme } from '@mui/system';
+import { ResponsiveLine } from '@nivo/line';
+import { linearGradientDef } from '@nivo/core';
+import { FormattedTime } from 'react-intl';
+
+import EmailLayout from 'features/emails/layout/EmailLayout';
+import { PageWithLayout } from 'utils/types';
+import { scaffold } from 'utils/next';
+import useEmail from 'features/emails/hooks/useEmail';
+import useEmailStats from 'features/emails/hooks/useEmailStats';
+import { useNumericRouteParams } from 'core/hooks';
+import useServerSide from 'core/useServerSide';
+import ZUICard from 'zui/ZUICard';
+import ZUINumberChip from 'zui/ZUINumberChip';
+import EmailKPIChart from 'features/emails/components/EmailKPIChart';
+import useEmailInsights from 'features/emails/hooks/useEmailInsights';
+import ZUIFuture from 'zui/ZUIFuture';
+import { useMessages } from 'core/i18n';
+import messageIds from 'features/emails/l10n/messageIds';
+
+export const getServerSideProps: GetServerSideProps = scaffold(
+  async () => {
+    return {
+      props: {},
+    };
+  },
+  {
+    authLevelRequired: 2,
+    localeScope: ['layout.organize.email', 'pages.organizeEmail'],
+  }
+);
+
+const EmailPage: PageWithLayout = () => {
+  const messages = useMessages(messageIds);
+  const { orgId, emailId } = useNumericRouteParams();
+  const theme = useTheme();
+  const { data: email } = useEmail(orgId, emailId);
+  const stats = useEmailStats(orgId, emailId);
+  const insightsFuture = useEmailInsights(orgId, emailId);
+
+  const onServer = useServerSide();
+
+  if (onServer || !email) {
+    return null;
+  }
+
+  return (
+    <>
+      <Head>
+        <title>{email.title}</title>
+      </Head>
+      <ZUICard
+        header={messages.insights.opened.header()}
+        status={
+          <ZUINumberChip
+            color={theme.palette.grey[200]}
+            value={stats.numOpened}
+          />
+        }
+      >
+        <Box display="flex" gap={1}>
+          <Box flexGrow={0} maxWidth={300}>
+            <EmailKPIChart
+              email={email}
+              title={messages.insights.opened.gauge.header()}
+              total={stats.numSent}
+              value={stats.numOpened}
+            />
+            <Typography mb={2} mt={1} variant="body2">
+              {messages.insights.opened.description()}
+            </Typography>
+          </Box>
+          <Box flexGrow={1} height={550}>
+            <ZUIFuture future={insightsFuture}>
+              {(insights) => (
+                <ResponsiveLine
+                  animate={false}
+                  axisBottom={{
+                    format: '%b %d',
+                  }}
+                  colors={[theme.palette.primary.main]}
+                  curve="basis"
+                  data={[
+                    {
+                      data: insights.opensByDate.map((openEvent) => ({
+                        x: new Date(openEvent.date),
```

Comment:
Because `getEmailInsights()` is code being run on the server (an RPC), and the code you're commenting on is being run on the client, so the data needs to be serializable between the two.